units/sysinit.target

```diff
@@ -10,6 +10,8 @@
 [Unit]
 Description=System Initialization
 Documentation=man:systemd.special(7)
-Conflicts=emergency.service emergency.target
+
 Wants=local-fs.target swap.target
-After=local-fs.target swap.target emergency.service emergency.target
+After=local-fs.target swap.target
+Conflicts=emergency.service emergency.target
+Before=emergency.service emergency.target
```

Comment:
this changes from `After` to `Before`, was that intended?

Reply:
which of the two order direction orders you pick doesn't really matter for Conflicts= lines, since if two units that have an ordering dep against each other are queued so that one shall be stopped and the other started, then we'll always do the stop first, the start second, rgeardless in which direction the ordering points.

i.e. the rule is for ordering jobs between units A and B:

1. if no ordering is defined between unit A and B run jobs in parallel
2. otherwise, if one job is start the other is stop, do the stop first, start second
3. otherwise, if both jobs are start, then run the job first whose unit has Before= on the other
4. otherwise, if both jobs are stop, then run the job first whose unit has After= on the other.

Now, given that we have Conflicts= here, we know that the only way how these units will both be scheduled in the job queue the same time is if one is start and the other is stop, hence the direction of the order dep doesn't matter, as long as it is there.

That said, it makes sense to be systematic here, since the order cycle checker actually just checks the order graph, and doesn't take start/stop stuff into account.

Reply:
We generally start `sysinit.target`, and then transition to `emergency.target` on error, so I think having `Before` is more logical.